tests/lib/openstack_query/runners/test_query_runner.py

```diff
@@ -0,0 +1,92 @@
+import unittest
+from unittest.mock import MagicMock, patch
+
+from openstack_query.runners.query_runner import QueryRunner
+
+
+class QueryRunnerTests(unittest.TestCase):
+    @patch.multiple(QueryRunner, __abstractmethods__=set())
+    def setUp(self):
+        """
+        Setup for tests
+        """
+        super().setUp()
+        self.mocked_connection = MagicMock()
+        self.instance = QueryRunner(connection_cls=self.mocked_connection)
+        self.conn = self.mocked_connection.return_value.__enter__.return_value
+
+    @patch("openstack_query.runners.query_runner.QueryRunner._apply_filter_func")
+    @patch("openstack_query.runners.query_runner.QueryRunner._run_query")
+    def test_run_only_filter_func(self, mock_run_query, mock_apply_filter_func):
+        mock_run_query.return_value = ["openstack-resource-1", "openstack-resource-2"]
+        mock_apply_filter_func.return_value = ["filtered-openstack-resource"]
+
+        res = self.instance.run(
+            cloud_account="test",
+            filter_func="some-filter-func",
```

Comment:
Pass a mock here, rather than a string
Then we can also assert that it's not called at this level (until we want it to be called)